pkg/ext/certs.go

```diff
@@ -0,0 +1,22 @@
+package ext
+
+import (
+	"fmt"
+
+	"k8s.io/apiserver/pkg/server/dynamiccertificates"
+	certutil "k8s.io/client-go/util/cert"
+)
+
+func certForCommonName(cname string) (dynamiccertificates.SNICertKeyContentProvider, error) {
+	certPem, keyPem, err := certutil.GenerateSelfSignedCertKey(cname, nil, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate self signed cert: %w", err)
+	}
+
+	content, err := dynamiccertificates.NewStaticSNICertKeyContent("self-signed-imperative", certPem, keyPem, cname)
```

Comment:
Have we found a way to get certificates that rotate? I'm not sure what the default expiry date is here, but we should ensure that the certificate is rotated before it expires, otherwise the main apiserver will no longer be able to communicate with Rancher.

Reply:
Its definitely possible, essentially we'd just need to implement a [`SNICertKeyProvider`](https://github.com/kubernetes/apiserver/blob/6885e995e148b2b66cee033eba554fc6f237366c/pkg/server/dynamiccertificates/interfaces.go#L62) that'll just spin up a new cert a little before the other one is set to expire. I'm not seeing anything about the expiry dates in the [`tls`](https://pkg.go.dev/crypto/tls) docs so I'll need to do some digging to make sure that they do expire.

Do we want the certs rotating in this PR or can we get this one in and leave the rotating certs for a following one?

Reply:
We can leave as is for this PR so that other folks can at least try out the imperative API with the setting enabled.

Let's make sure to get that in though for 2.11.